src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/transformer/PersonalTimelineTransformer.kt

```diff
@@ -32,4 +41,41 @@ class PersonalTimelineTransformer(
     createdBy = userTransformer.transformJpaToApi(application.createdByUser, ServiceName.approvedPremises),
     timelineEvents = timelineEvents,
   )
+
+  private fun transformApplication(
+    application: OfflineApplicationEntity,
+    timelineEvents: List<TimelineEvent>,
+  ) = ApplicationTimeline(
+    id = application.id,
+    createdAt = application.createdAt.toInstant(),
+    status = null,
+    isOfflineApplication = true,
+    createdBy = null,
+    timelineEvents = timelineEvents,
+  )
+}
+
+data class ApplicationTimelineModel(
+  val application: BoxedApplication,
+  val timelineEvents: List<TimelineEvent>,
+)
+
+sealed interface BoxedApplication {
```

Comment:
It's a shame that OfflineApplicationEntity a subtype of the ApplicationEntity, i'm sure there were reasons why and this is a nice way to reintroduce some polymorphism